File: sol/2294/2294.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)
        res = 1
        start = nums[0]
        for x in nums:
            if x - start > k:
                res += 1
                start = x
        return res

File: sol/2294/2294.cpp
struct Solution {
    int partitionArray(vector<int>& nums, int k) {
        ranges::sort(nums);
        int n = nums.size(), res = 1, start = nums[0];
        for (int x : nums){
            if (x > start + k){
                ++res;
                start = x;
            }
        }
        return res;
    }
};
